### DISCUSSION TITLE
Configure an "simple" IP Auth based libresbc

### DISCUSSION THREAD

**Discussion by khorsmann** [hnimminh/libresbc#23] [Q&A] [answered]
Hi there,
After reading the fine wiki i got more and more the problem - an working example is hard to adapt. 
Cause the webgui is not there, i created many json files and the steps to generate an flow.

And at the end - its not IP based, its "407 Proxy-Auth" from internal kamailio.....
How we get there more "easyness" to that?

For example

Public-IP (User) ---> Public IP (libresbc) ---> Private IP (libresbc) ----> Private IP (test-pbx/carrier).

First - the order to get this running was also not so easy. Try and error:

```
curl -k -X POST -H "Content-Type: application/json" -d @./internal-netalias.json https://192.168.122.253:8443/libreapi/base/netalias
curl -k -X POST -H "Content-Type: application/json" -d @./internal-sipprofile.json https://192.168.122.253:8443/libreapi/sipprofile

curl -k -X POST -H "Content-Type: application/json" -d @./external-netalias.json https://192.168.122.253:8443/libreapi/base/netalias
curl -k -X POST -H "Content-Type: application/json" -d @./external-sipprofile.json https://192.168.122.253:8443/libreapi/sipprofile

curl -k -X POST -H "Content-Type: application/json" -d @./access-domain.json https://192.168.122.253:8443/libreapi/access/domain-policy
curl -k -X POST -H "Content-Type: application/json" -d @./access-service.json https://192.168.122.253:8443/libreapi/access/service

curl -k -X POST -H "Content-Type: application/json" -d @./media_class_pcma.json https://192.168.122.253:8443/libreapi/class/media
curl -k -X POST -H "Content-Type: application/json" -d @./gateway_internal.json https://192.168.122.253:8443/libreapi/base/gateway

curl -k -X PUT -H "Content-Type: application/json" -d @./default_global_cluster.json https://192.168.122.253:8443/libreapi/cluster
curl -k -X POST -H "Content-Type: application/json" -d @./default_capacity.json https://192.168.122.253:8443/libreapi/class/capacity
curl -k -X POST -H "Content-Type: application/json" -d @./interconnection_outbound.json https://192.168.122.253:8443/libreapi/interconnection/outbound

curl -k -X POST -H "Content-Type: application/json" -d @./default_inbound_routing.json https://192.168.122.253:8443/libreapi/routing/table

# here
###curl -k -X POST -H "Content-Type: application/json" -d @./default_in_routing_record.json https://192.168.122.253:8443/libreapi/routing/record
curl -k -X POST -H "Content-Type: application/json" -d @./interconnection-inbound.json https://192.168.122.253:8443/libreapi/interconnection/inbound
curl -k -X PATCH -H "Content-Type: application/json" -d @./firewall_whiteset.json https://192.168.122.253:8443/libreapi/base/firewall/whiteset
```

**Comment by hnimminh** [comment]
Hi @khorsmann, 
If that just a connection to provider or peer, consider to use the `Outbound/Inbound Interconnection`
The IP Auth is was not support for access layer yet, at the time, I leave and waiting for someone file a feature request for it. I willing to add

I know that configuration via `curl` is not _easy_, use the GUI tool like `postman` with import openapi from `https://192.168.122.253:8443/apidocs`  may get more easier. 
For error, can you show me the error, so that I can comment on the issue, if there is a private info, you can send me the email.

**Comment by khorsmann** [comment]
Hi @hnimminh 

some config examples would be nice.
I know, there is no "one way to do it" - but it seems there is a @hnimminh libresbc way to configure it.

What did you mean with "Outbound/Inbound Interconnection"?

**Reply by redand65** [reply]
hi@khorsmann,
i am just trying to do a simple access sip trunk that gets routed to an asterisk pbx, and then a second session where the asterisk  can go offnet to a sip provider. Have you gotten that to work? I have not had much luck wiht the apidocs...

**Comment by hnimminh** [comment, marked as answer]
Public-IP (User) ---> Public IP (libresbc) ---> Private IP (libresbc) ----> Private IP (test-pbx/carrier).

Here are some summary steps need to be done on LibreSBC

1. Define `cluster node`, `netalias`
2. Define 2 SIP profile `1 for public` and `1 for private` 
3. Define `Media Class`, `Capacity Class`
4. For `Private IP (libresbc) ----> Private IP (test-pbx/carrier).` = `libreapi/interconnection/outbound` + `libreapi/base/gateway`
5. Routing Table: `/libreapi/routing/table` with `action=route` and `routes=<outbound-connection-in-step-4>`
6. For `Public-IP (User) ---> Public IP (libresbc)` = `libreapi/interconnection/inbound` 

Hope this help.

`some config examples would be nice.` thanks for suggestion. I will try to create an example in wiki in this/next weekend once i have time.

**Comment by khorsmann** [comment]
Hi @hnimminh,

thanks for the hints. I try to configure it that way. A question to step 2 - sipprofile context is enum "core", "carrier", "access" - in this example all profiles got "carrier" i assume?

**Reply by hnimminh** [reply]
actually, any of them will work for your scenario.
